packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2911,15 +2911,14 @@ class BuildOwner {
                   'The dirty list for the current build scope is: ${buildScope._dirtyElements}',
         );
       }
-      // When reactivating an inactivate Element, _scheduleBuildFor should only be
-      // called within _flushDirtyElements.
       if (!_debugBuilding && element._inDirtyList) {
```

Comment:
This check assume repeated call happens within the same frame, which is no longer the case in the layoutbuilder.

WDYT about updating the check to take into account that it is not guarantee the build will be flush every frame.

Reply:
One possible way is to somehow only enforce this on root build scope

Reply:
actually know I think of it the current pr makes more sense. there is just nothing to do for rebuild for layoutBuilder

Reply:
Ideally the assert should still hold: https://github.com/flutter/flutter/pull/154681